public/js/navigator-js/integration/ViewRecipe.js

```diff
@@ -38,6 +42,15 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 		toView: function(viewClass) {
 			this._viewClass = viewClass;
 
+			// This is where the recipe adds a mixin depending on the type of viewClass
+			// added.  This allows for different handling for different 'types' of
+			// view recipes.
+
+			// TODO: Find a more secure way to determine whether class
+			// extends react component
+
```

Comment:
You might be able to do something like this. I test it with a simple class that just extended Backbone.View. I imagine it would work with React.Component was well.
```
Backbone.View.prototype.isPrototypeOf(viewClass.prototype)
```